cmake/BLTPrivateMacros.cmake

```diff
@@ -842,3 +842,145 @@ macro(blt_clean_target)
+    if(_BLT_${_target_upper}_IS_REGISTERED_LIBRARY)
+        set(_is_blt_registered_target TRUE)
+        message (STATUS "[${arg_TARGET} property] '${arg_TARGET}' is a blt_registered target")
+    endif()
+
+
+    if(_is_cmake_target)
+        ## Solution adapted from https://stackoverflow.com/q/32183975
+        ## Create list of cmake properties
+        set(_property_list)
+        execute_process(COMMAND cmake --help-property-list OUTPUT_VARIABLE _property_list)
+        string(REGEX REPLACE ";" "\\\\;" _property_list "${_property_list}")
+        string(REGEX REPLACE "\n" ";" _property_list "${_property_list}")
+        blt_filter_list(TO _property_list REGEX "^LOCATION$|^LOCATION_|_LOCATION$" OPERATION "exclude")
+        blt_list_remove_duplicates(TO _property_list)   
+
+        ## For interface targets, filter against whitelist of valid properties
+        get_property(_target_type TARGET ${arg_TARGET} PROPERTY TYPE)
+        if("${_target_type}" STREQUAL "INTERFACE_LIBRARY")
+            blt_filter_list(TO _property_list
+                            REGEX "^(INTERFACE_|IMPORTED_LIBNAME_|COMPATIBLE_INTERFACE_|MAP_IMPORTED_CONFIG_)|^(NAME|TYPE|EXPORT_NAME)$"
+                            OPERATION "include")
+        endif()
+
+        ## Print all such properties that have been SET
+        foreach (prop ${_property_list})
+            string(REPLACE "<CONFIG>" "${CMAKE_BUILD_TYPE}" prop ${prop})
+            get_property(_propval TARGET ${arg_TARGET} PROPERTY ${prop} SET)
+            if ("${_propval}" AND "${prop}" MATCHES "${arg_PROP_REGEX}")
+                get_target_property(_propval ${arg_TARGET} ${prop})
+                message (STATUS "[${arg_TARGET} property] ${prop}: ${_propval}")
+            endif()
+        endforeach()
+        unset(_property_list)
+        unset(_propval)
+    endif()
+
+    ## Additionally, output variables generated via blt_register_target of the form "_BLT_<target>_*"
+    if(_is_blt_registered_target)
+        set(_target_prefix "_BLT_${_target_upper}_")
+
+        ## Filter to get variables of the form _BLT_<target>_ and print
+        get_cmake_property(_variable_names VARIABLES)
+        foreach (prop ${_variable_names})
+            if("${prop}" MATCHES "^${_target_prefix}" AND "${prop}" MATCHES "${arg_PROP_REGEX}")
+                message (STATUS "[${arg_TARGET} property] ${prop}: ${${prop}}")
+            endif()
+        endforeach()
+        unset(_target_prefix)
+        unset(_variable_names)
+    endif()
+
+    unset(_target_upper)
+    unset(_is_blt_registered_target)
+    unset(_is_cmake_target)
+endmacro(blt_print_target_properties_private)
+
+##------------------------------------------------------------------------------
+## blt_find_all_targets_recursive(TARGET <target> TLIST tlist)
+##
+## Store all target's dependancies (link libraries and interface link libraries)
+## recursively in TLIST.
+##
+## NOTE: TLIST should be an empty string at the start. TLIST must be called tlist
```

Comment:
Can you remove this requirement?

`if(NOT DEFINED tlist)` -> `if(NOT DEFINED ${arg_TLIST})`
`list(APPEND tlist ${t})` -> `list(APPEND ${arg_TLIST} ${t})`

Reply:
yes that sounds much better